sp/src/game/client/c_baselesson.cpp

```diff
@@ -1039,40 +1040,40 @@ Vector CIconLesson::GetIconTargetPosition( C_BaseEntity *pIconTarget )
 
 #define LESSON_VARIABLE_INIT_SYMBOL( _varEnum, _varName, _varType ) g_n##_varEnum##Symbol = KeyValuesSystem()->GetSymbolForString( #_varEnum );
 
-#define LESSON_SCRIPT_STRING_ADD_TO_MAP( _varEnum, _varName, _varType ) g_NameToTypeMap.Insert( #_varEnum, LESSON_VARIABLE_##_varEnum## );
```

Comment:
A note to the changes here: The `##` operator is for creating a *singular* token. In this instance, for example, pasting `LESSON_VARIABLE_` with `_varEnum` with `);` will never create a valid *singular* token. Here three tokens are to be created: `LESSON_VARIABLE_##_varEnum` `)` `;`.
gcc is more strict than MSVC here in that it rejects the operation if pasting fails.